### PR TITLE
NodeExporter: Added ZFS toggle (defaults to off), fixes #1930

### LINKED ISSUE #1930: os-node_exporter collector failed  for zfs with err="couldn't get sysctl: no such file or directory"
After updating my OPNsense installation to version 20.1.8_1-amd64 I constantly get the following error message in system.log:

```
Jul 21 00:36:59 opnsense node_exporter[88175]: level=error ts=2020-07-20T22:36:59.251Z caller=collector.go:161 msg="collector failed" name=zfs duration_seconds=1.4879e-05 err="couldn't get sysctl: no such file or directory" 
```

Software versions are as follows:
OPNsense 20.1.8_1-amd64
FreeBSD 11.2-RELEASE-p20-HBSD
OpenSSL 1.1.1g 21 Apr 2020

node_exporter version=1.0.1, branch=release-1.0, revision=0

The startup log of node_exporter is

```
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.550Z caller=node_exporter.go:177 msg="Starting node_exporter" version="(version=1.0.1, branch=release-1.0, revision=0)" 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.550Z caller=node_exporter.go:178 msg="Build context" build_context="(go=go1.14.4, user=root, date=)" 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.550Z caller=node_exporter.go:105 msg="Enabled collectors" 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=boottime 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=cpu 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=devstat 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=exec 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=filesystem 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=loadavg 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=meminfo 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=netdev 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=textfile 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=time 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=uname 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:112 collector=zfs 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=node_exporter.go:191 msg="Listening on" address=0.0.0.0:9100 
Jul 21 00:36:55 opnsense node_exporter[88175]: level=info ts=2020-07-20T22:36:55.551Z caller=tls_config.go:170 msg="TLS is disabled and it cannot be enabled on the fly." http2=false 
Jul 21 00:36:56 opnsense node_exporter[88175]: level=error ts=2020-07-20T22:36:56.268Z caller=collector.go:161 msg="collector failed" name=zfs duration_seconds=2.6973e-05 err="couldn't get sysctl: no such file or directory" 
Jul 21 00:36:57 opnsense node_exporter[88175]: level=error ts=2020-07-20T22:36:57.251Z caller=collector.go:161 msg="collector failed" name=zfs duration_seconds=1.555e-05 err="couldn't get sysctl: no such file or directory" 
Jul 21 00:36:58 opnsense node_exporter[88175]: level=error ts=2020-07-20T22:36:58.251Z caller=collector.go:161 msg="collector failed" name=zfs duration_seconds=1.7693e-05 err="couldn't get sysctl: no such file or directory" 
```

### PR DESCRIPTION
**Pull request by jkellerer**
Fixes #1930 

When ZFS is disabled (= default) the following log is silenced:
````
..  msg="collector failed" name=zfs duration_seconds=1.4879e-05 err="couldn't get sysctl: no such file or directory"
````
This change also fixes toggling collectors that are enabled by default (see https://github.com/prometheus/node_exporter): 
cpu, exec, filesystem, loadavg, meminfo, netdev, time

Not fixed: 
* "interrupts" collector is implemented for OpenBSD not FreeBSD. It cannot be toggled unless node_exporter implements it.
* I've not checked if ZFS collector is generally broken. Since the *Telegraf* plugin has a ZFS toggle, I did not remove ZFS completely.

### CODE REVIEW

File: sysutils/node_exporter/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 PLUGIN_NAME=		node_exporter
-PLUGIN_VERSION=		1.0
+PLUGIN_VERSION=		1.1
 PLUGIN_REVISION=	1
```
Comment (fichtner) [inline sysutils/node_exporter/Makefile:3]:
```suggestion
```

File: sysutils/node_exporter/pkg-descr [hunk 2]
```diff
@@ -2,3 +2,33 @@ Prometheus exporter for hardware and OS metrics exposed by *NIX kernels,
...
+* Allow to toggle collectors:
+    cpu
+    exec
+    filesystem
+    loadavg
+    meminfo
+    netdev
+    ntp
+    time
+    devstat
```
Comment (fichtner) [inline sysutils/node_exporter/pkg-descr:34]: very nice, thanks

File: sysutils/node_exporter/src/opnsense/mvc/app/models/OPNsense/NodeExporter/General.xml [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <description>
         node_exporter - Prometheus exporter for hardware and OS metrics.
     </description>
-    <version>0.1.0</version>
+    <version>0.2.0</version>
```
Comment (fichtner) [inline sysutils/node_exporter/src/opnsense/mvc/app/models/OPNsense/NodeExporter/General.xml:6]: We only need this if new default values are added

```suggestion
    <version>0.1.0</version>
```

File: sysutils/node_exporter/Makefile [hunk 4]
```diff
@@ -1,6 +1,5 @@
 PLUGIN_NAME=		node_exporter
-PLUGIN_VERSION=		1.0
-PLUGIN_REVISION=	1
+PLUGIN_VERSION=		1.1
 PLUGIN_COMMENT=		Prometheus exporter for machine metrics
 PLUGIN_DEPENDS=		node_exporter
 PLUGIN_MAINTAINER=	dharrigan@gmail.com
```
Comment (fichtner) [inline sysutils/node_exporter/Makefile:5]: @jkellerer are you interested in becoming the maintainer? David is busy elsewhere it seems.

Comment (jkellerer) [reply sysutils/node_exporter/Makefile:5]: @fichtner, yes sure. That’s fine for me.

Comment (fichtner) [reply sysutils/node_exporter/Makefile:5]: ok, just replace mail address in that maintainer line via commit

Comment (jkellerer) [reply sysutils/node_exporter/Makefile]: done

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**